src/libaktualizr/http/httpclient.cc

```diff
@@ -81,6 +81,14 @@ HttpClient::HttpClient(const std::vector<std::string>* extra_headers,
   if (curl == nullptr) {
     throw std::runtime_error("Could not initialize curl");
   }
+  share = curl_share_init();
+  if (share == nullptr) {
+    throw std::runtime_error("Could not initialize share");
+  }
+  curl_share_setopt(share, CURLSHOPT_SHARE, CURL_LOCK_DATA_SSL_SESSION);
+  curl_share_setopt(share, CURLSHOPT_SHARE, CURL_LOCK_DATA_DNS);
+  curl_share_setopt(share, CURLSHOPT_SHARE, CURL_LOCK_DATA_CONNECT);
```

Comment:
It looks like `CURL_LOCK_DATA_CONNECT` requires synchronization in a multi-threaded env, while `CURL_LOCK_DATA_SSL_SESSION` does not require synchronization.

If my assumption is correct then, maybe it's suffice to re-use just mTLS context.